src/core/stores/VDocument.ts

```diff
@@ -27,21 +27,37 @@ export class VDocument {
     /**
      * Insert text at range.
      *
+     * If the range is collapsed, add `text` to the vDocument and copy the formating of the
+     * previous char or the next char.
+     *
+     * If the range is not collapsed, replace the text with the formating that was present in the
+     * range.
+     *
      * @param text
      */
     insertText(text: string): void {
-        // Remove the contents of the selection if needed.
-        if (!this.range.isCollapsed()) {
+        let format: FormatType = {};
+        if (this.range.isCollapsed()) {
+            const charToCopyFormat = this.range.start.previousCharOnParent() ||
+                this.range.start.nextCharOnParent() || { format: {} };
+            format = { ...charToCopyFormat.format, ...this.range.format };
```

Comment:
move this logic to editor.getCurrentFormat()

Reply:
To answer the question: it is dynamic because the value of range.format contain only the key where we want to force the next format. So by default, it uses the charToCopyFormat.format, and override the key that we want to force with ...this.range.format